tests/integration/test_asynchronous_metrics_pk_bytes_fields/test.py

```diff
@@ -0,0 +1,57 @@
+import time
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+node1 = cluster.add_instance(
+    "node1",
+    with_zookeeper=True,
+    main_configs=["configs/asynchronous_metrics_update_period_s.xml"],
+)
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def test_total_pk_bytes_in_memory_fields(started_cluster):
+    try:
+        cluster.start()
+        node1.query("SET log_queries = 1;")
+        node1.query("CREATE DATABASE replica;")
+        query_create = """CREATE TABLE replica.test
+        (
+           id Int64,
+           event_time DateTime
+        )
+        Engine=MergeTree()
+        PARTITION BY toYYYYMMDD(event_time)
+        ORDER BY id;"""
+        time.sleep(2)
+        node1.query(query_create)
+        node1.query("""INSERT INTO replica.test VALUES (1, now())""")
+        node1.query("SYSTEM FLUSH LOGS;")
+
+        # query system.asynchronous_metrics
+        test_query = "SELECT count() > 0 ? 'ok' : 'fail' FROM system.asynchronous_metrics WHERE metric='TotalPrimaryKeyBytesInMemory';"
```

Comment:
I suspect there is a race between query and `asynchronous_metrics_update_period_s`

Reply:
Interesting, what are you observing?

Reply:
My suspicions is the following:
- async metrics calc new values time to time
-  `SYSTEM FLUSH LOGS` flushes the current values to the logs, they could be stale
To overcome this, you need to do retries with the series `flush + select`.

Also I suspect that the values are not zero even without insertion. Some logs tables could have `MergeTree` strorage, so they all add some value to the sum.
To overcome this, you could do the following:
- ask metric before insertion.
- make table index with granularity 1
- insert 1m rows
- check that you counters have raise accordingly

Reply:
Ah I see. I actually wrote the tests to be simple and just test if the fields get populated without error and so it tests nothing more.

But I think your suggestion makes sense. I will probably write a test that checks the value before and after and asserts if it's greater after inserting data and lesser after deleting data (couldn't probably assert the exact value, but asserting for > or < would work fine) and I will add retries.

Reply:
Added the test. `asynchronous_metrics_update_period_s` works for higher values too (for very high values need to override the current default timeout and retries in the test) but setting it to 1 for the purposes of CI (similar to other such integration tests).

Reply:
In general it is all good. I have left some little remarks about code in the test.

Now, when you test is good and survive long `update_period_s`, it is ready to by just `.sh` stateless test.
Stateless tests are chipper. Even when the test is waiting for a minute or two. When stateless test is waiting it consume only one process in CI. A lot of them are run in parallel. Integration test is heavy due to starting new instance of server in separate docker container.
This is optional. I do not insist.

Reply:
Thank you for your reviews. I've addressed the remarks. I do agree that stateless tests are lighter, but for testing system metrics I often favor doing it via integration tests as the tests are more isolated.

Having said that,  can I try your suggestion as a follow up later? I can work on converting it to a stateless test and open a PR later. It could possibly work well as a stateless test too.